### PR TITLE
Documentation smart snippet

### PR DESCRIPTION
**Pull request by olamothe**
https://coveord.atlassian.net/browse/JSUI-3230

[Deploy](https://dashboard.heroku.com/pipelines/a3535101-5bbf-4a5b-a909-47fcf8c9f149)

### CODE REVIEW

**Comment by olamothe** [comment]
@rtherien : I think you will be in charge of documentation the question answering feature on docs.coveo.com

I added you as a reviewer here, since you will see that it's * extremely * succinct ;) 

I'm going to wait for you to give comments/suggestions, since the wording and all should correspond with what will happen on docs.coveo.com.

We will most likely want to also add a more detailed section in the documentation about the UI components themselves, especially how you can use inline CSS to control the excerpt styles.

**Review by ThibodeauJF — APPROVED** [review]

**Review by samisayegh — APPROVED** [review]

File: src/ui/SmartSnippet/SmartSnippet.ts [hunk 1]
```diff
@@ -74,7 +74,11 @@ export const SmartSnippetClassNames = {
 export interface ISmartSnippetOptions {
   maximumSnippetHeight: number;
 }
-
+/**
+ * SmartSnippet allows a to display the excerpt of a document that would be the most likely to answer a particular query.
```
Comment (samisayegh) [inline src/ui/SmartSnippet/SmartSnippet.ts:78]: `SmartSnippet allow displaying ...`?

File: src/ui/SmartSnippet/SmartSnippetSuggestions.ts [hunk 2]
```diff
@@ -23,6 +23,9 @@ export const SmartSnippetSuggestionsClassNames = {
   QUESTIONS_LIST_TITLE_CLASSNAME
 };
 
+/**
+ * SmartSnippetSuggestions allows to display recommended related suggestions to end users.
```
Comment (samisayegh) [inline src/ui/SmartSnippet/SmartSnippetSuggestions.ts:27]:
```
The SmartSnippetSuggestions component displays recommended ...
```

**Review by lvu285 — APPROVED** [review]

**Comment by olamothe** [comment]
> When you refer to "add a more detailed section in the documentation about the UI components themselves, especially how you can use inline CSS to control the excerpt styles." are you referring to the documentation in this PR (what will appear on coveo.github.io), or to the documentation that will appear on docs.coveo.com?

I was thinking that we would add an articule on docs.coveo.com that explains how you can customize the UI components/the snippet using CSS. 

And that we would want to add, inside Github reference doc, a link to that article that goes more in depth.

**Comment by rtherien** [comment]
> > When you refer to "add a more detailed section in the documentation about the UI components themselves, especially how you can use inline CSS to control the excerpt styles." are you referring to the documentation in this PR (what will appear on coveo.github.io), or to the documentation that will appear on docs.coveo.com?
> 
> I was thinking that we would add an articule on docs.coveo.com that explains how you can customize the UI components/the snippet using CSS.
> 
> And that we would want to add, inside Github reference doc, a link to that article that goes more in depth

Sounds good to me!

**Pull request merged**